youtube_dl/downloader/external.py

```diff
@@ -27,15 +27,20 @@ def real_download(self, filename, info_dict):
 
         retval = self._call_downloader(tmpfilename, info_dict)
         if retval == 0:
-            fsize = os.path.getsize(encodeFilename(tmpfilename))
-            self.to_screen('\r[%s] Downloaded %s bytes' % (self.get_basename(), fsize))
-            self.try_rename(tmpfilename, filename)
-            self._hook_progress({
-                'downloaded_bytes': fsize,
-                'total_bytes': fsize,
-                'filename': filename,
-                'status': 'finished',
-            })
+            if filename == '-':
+                # xxx report_progress() expects total_bytes to be set or it throws a KeyError, so
```

Comment:
Thanks for finding out the bug. However, progress hooks should always be called; otherwise third party applications using youtube-dl's Python API will be broken. In this case ```report_progress``` should be fixed instead.